python-spec/src/somacore/coordinates.py

```diff
@@ -0,0 +1,470 @@
+"""Definitions of types related to coordinate systems."""
+
+import abc
+import collections.abc
+from dataclasses import dataclass
+from typing import Any, Optional, Sequence, Tuple, Union
+
+import numpy as np
+import numpy.typing as npt
+
+
+@dataclass
```

Comment:
Historically we have used `attrs` over `dataclasses` in the SOMA API surface (I believe it is only used in a few places at the moment). This was based on the recommendation of @thetorpedodog 

Is there a compelling reason to use `dataclasses` here, vs `attrs`?  My hot take is we should pick one and stick with it across the entire API.

Reply:
yikes

i totally missed seeing that -- but bruce is quite right

i wish we had a way to autodetect this ... maybe a ci script that greps for '@dataclass' -- ?

Reply:
@jp-dark - I see the same pattern in other PRs (e.g., #219). I won't note those, but ideally we would apply the same (TBD) policy to all of them.  I'm personally quite comfortable with using `attrs`, but happy to chat about it if there is a specific motivation to introduce `dataclasses`